### PR TITLE
Supporting more rules for egress

### LINKED ISSUE #164: Cannot treat Security Group egress in the same way as we do with ingress
### Describe the Bug

When configuring security group ingress I can specify either a list of CIDR blocks, or an additional security group.
With egress, instead, I can only either disable it or have it fully open (any port, any protocol, 0.0.0.0/0)

### Expected Behavior

Being able to specify CIDR and security groups for egress as well

### Steps to Reproduce

N/A

### Screenshots

_No response_

### Environment

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by vale21**
## what

Support the ability to specify CIDR blocks and additional security groups for egress as well

## why

improve security groups

## references

Fixes #164

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -55,17 +55,28 @@ resource "aws_security_group_rule" "ingress_cidr_blocks" {
...
   from_port         = 0
   to_port           = 0
   protocol          = "-1"
-  cidr_blocks       = ["0.0.0.0/0"]
+  cidr_blocks       = var.egress_allowed_cidr_blocks
   security_group_id = join("", aws_security_group.default.*.id)
 }
 
+resource "aws_security_group_rule" "egress_security_groups" {
+  count                    = local.enabled && var.egress_enabled ? length(var.security_groups) : 0
```
[the remark this answers is not in the document]
Comment (vale21) [reply main.tf]: It's not. Fixed

**Comment by vale21** [comment]
Thank you @aknysh. Changes integrated

**Review by Nuru — DISMISSED** [review]

**Comment by vale21** [comment]
Hello guys. Any update on this? Is something still pending? Can it be merged?

**Comment by vale21** [comment]
@aknysh could you plese review the changes that I pushed? This PR is blocked since a lot of time

**Comment by vale21** [comment]
@aknysh @Nuru guys, I incoprporated the requested changes. can I have a review please?

**Comment by taliesins** [comment]
@aknysh and @Nuru this seems like an easy win. What do we need to do to get this merged in? I believe @vale21 has made all the requested changes.

**Comment by vale21** [comment]
> @vale21 thanks for the PR.
> 
> Please see the comments. Please also format all TF code by running `terraform fmt`. And also, please run the following commands:
> 
> ```
> make init
> make github/init
> make readme
> ```
> 
> and commit the changes
> 
> thanks

Hi @aknysh, this has been done a while ago. Can you review please?

**Comment by hans-d** [comment]
@aknysh can you re review?

**Pull request closed without merging**